pkgs/misc/emulators/tilem/default.nix

```
@@ -0,0 +1,66 @@
+{ stdenv, fetchurl, runCommand, lib, libarchive, autoreconfHook,
+  pkgconfig, glib, libusb1, darwin, acl, lzma, bzip2, gnome2 }:
+
+let
+  tilibs = fetchurl {
+    url = "https://www.ticalc.org/pub/unix/tilibs.tar.gz";
+    sha256 = "07cfwwlidgx4fx88whnlch6y1342x16h15lkvkkdlp2y26sn2yxg";
+  };
+  tilibs2 = runCommand "tilibs2" {} ''
+    mkdir $out
+    cd $out
+    unpackFile ${tilibs}
+    mv -vi tilibs2/* .
+    rmdir tilibs2
+  '';
+  libticonv = stdenv.mkDerivation {
+    pname = "libticonv";
+    version = "1.1.5";
+    src = "${tilibs2}/libticonv-1.1.5.tar.bz2";
+    nativeBuildInputs = [ autoreconfHook pkgconfig ];
+    buildInputs = [ glib ];
+    configureFlags = [ "--enable-iconv" ];
+  };
+  libticables = stdenv.mkDerivation {
+    pname = "libticables";
+    version = "1.3.5";
+    src = "${tilibs2}/libticables2-1.3.5.tar.bz2";
+    nativeBuildInputs = [ autoreconfHook pkgconfig ];
+    buildInputs = [ glib libusb1 ];
+    configureFlags = [ "--enable-libusb10" ];
+  };
+  libticalcs = stdenv.mkDerivation {
+    pname = "libticalcs";
+    version = "1.1.9";
+    src = "${tilibs2}/libticalcs2-1.1.9.tar.bz2";
+    nativeBuildInputs = [ autoreconfHook pkgconfig ];
+    buildInputs = [ glib libticables libticonv libtifiles lzma bzip2 ]
+                  ++ lib.optionals stdenv.isLinux [ acl ]
+                  ++ lib.optionals stdenv.isDarwin [ darwin.libobjc ];
+  };
+  libtifiles = stdenv.mkDerivation {
+    pname = "libtifiles";
+    version = "1.1.7";
+    src = "${tilibs2}/libtifiles2-1.1.7.tar.bz2";
+    nativeBuildInputs = [ autoreconfHook pkgconfig ];
+    buildInputs = [ glib libticonv libarchive lzma bzip2 ];
+  };
+in
+stdenv.mkDerivation {
+  name = "tilem";
+  version = "2.0";
+  src = fetchurl {
+    url = "https://sourceforge.net/projects/tilem/files/tilem/2.0/tilem-2.0.tar.bz2/download";
+    name = "tilem-2.0.tar.bz2";
+    sha256 = "1ba38xzhp3yf21ip3cgql6jzy49jc34sfnjsl4syxyrd81d269zw";
+  };
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ glib gnome2.gtk libticonv libtifiles libticables libticalcs ];
+  meta = with stdenv.lib; {
+    homepage    = "http://lpg.ticalc.org/prj_tilem/";
+    description = "Emulator and debugger for Texas Instruments Z80-based graphing calculators.
+";
```

```suggestion
    description = "Emulator and debugger for Texas Instruments Z80-based graphing calculators";
```
https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md